api/envoy/config/core/v3/grpc_service.proto

```diff
@@ -294,4 +294,9 @@ message GrpcService {
   // documentation on :ref:`custom request headers
   // <config_http_conn_man_headers_custom_request_headers>`.
   repeated HeaderValue initial_metadata = 5;
+
+  // Indicates the retry policy for re-establishing the gRPC channel
+  // This field is optional. If not provided, base interval is set to 500 ms, max interval set to 30 secs.
+  // If max interval is not provided, it will be set to ten times the base interval.
+  RetryPolicy retry_policy = 6;
```

Comment:
The issue with adding this field here is that there are other places that use `GrpcService` (i.e., non-xDS) that at the moment do not parse this field.

I wonder if this should be a property of the underlying client, and then one just configures the retry policy for EnvoyGrpc, and uses the [gRPC retries](https://github.com/grpc/proposal/blob/master/A6-client-retries.md) mechanism for GoogleGrpc. @htuch any opinions on this?

Reply:
Sorry, I am new to the project. 
Are you suggesting this is moved inside message EnvoyGrpc (https://github.com/envoyproxy/envoy/blob/bb7d941e7e245ebd89ca706d428d92f050d5e7b3/api/envoy/config/core/v3/grpc_service.proto#L32)?

Reply:
I think that EnvoyGrpc and GoogleGrpc will have different mechanism to handle backoff.
Specifically for this PR, I think you need to have the retry config in `EnvoyGrpc`, and the actual code that handles retries in [AsyncClientImpl](https://github.com/envoyproxy/envoy/blob/c394382139010828c125a23eccb47a5172e2eab8/source/common/grpc/async_client_impl.h#L23). An added value of this is that other uses of `AsyncClientImpl` (say filters that use gRPC), will also have that mechanism implemented.

The GoogleGrpc will have a different retry mechanism that is supported by the library.

Reply:
Thanks a lot for the feedback.

 It looks like adding the backoff mechanism to `AsyncClientImpl` would require extensive design changes. The goal of https://github.com/envoyproxy/envoy/issues/24236 and this PR was to resolve the TODO of making the  backoff parameters configurable. 

I added the `BackoffStrategy` to `ApiConfigSource` for this iteration,  I can raise another issue for refactoring `EnvoyGrpc` retry mechanism if that works. 

Thanks again.

Reply:
Thanks! Will it be possible to add the config into `EnvoyGrpc`. The motivation is to avoid the need to deprecate the change to the `ApiConfigSource` in the future?

/wait-any

Reply:
Sounds good, Moved the config into `EnvoyGrpc`.  Thanks